viper/parser/parser.py

```diff
@@ -282,55 +282,66 @@ def mk_full_signature(code):
     return o
 
 
+def parse_events(sigs, _events):
+    for event in _events:
+        sigs[event.target.id] = EventSignature.from_declaration(event)
+    return sigs
+
+
+def parse_external_contracts(external_contracts, _contracts):
+    for _contractname in _contracts:
+        _contract_defs = _contracts[_contractname]
+        _defnames = [_def.name for _def in _contract_defs]
+        contract = {}
+        if len(set(_defnames)) < len(_contract_defs):
+            raise VariableDeclarationException("Duplicate function name: %s" % [name for name in _defnames if _defnames.count(name) > 1][0])
+        for _def in _contract_defs:
+            sig = FunctionSignature.from_definition(_def)
+            contract[sig.name] = sig
+        external_contracts[_contractname] = contract
+    return external_contracts
```

Comment:
I like it. Just on thing: _contracts can probably be renamed to simple contracts now right?

Reply:
Do you mean changing `_contracts` to `contract?

Reply:
@DavidKnott nope I was just talking about the _contracts part, but now that I read it fully there are few variables we can reme the `_` prefix from :smile: 
```python
def parse_external_contracts(external_contracts, contracts):
    for contract_name in contracts:
        contract_defs = contracts[contract_name]
        _defnames = [_def.name for _def in contract_defs]
        contract = {}
        if len(set(_defnames)) < len(contract_defs):
            raise VariableDeclarationException("Duplicate function name: %s" % [name for name in _defnames if _defnames.count(name) > 1][0])
        for _def in contract_defs:
            sig = FunctionSignature.from_definition(_def)
            contract[sig.name] = sig
        external_contracts[contract_name] = contract
    return external_contracts
```